packages/nextcloud/lib/src/webdav/client.dart

```diff
@@ -210,24 +239,24 @@ class WebDavClient {
     File file, {
     void Function(double progress)? onProgress,
   }) async {
-    final sink = file.openWrite();
     final response = await getStream(path);
-    if (response.contentLength > 0) {
+    final contentLength = response.contentLength;
+    if (contentLength != null && contentLength > 0) {
+      final sink = file.openWrite();
```

Comment:
If the response is empty, we still want to write into the file.

Reply:
let's make a separate PR.
I just carried over the old approach

Reply:
But let's fix it later if needed.